src/service/git/octokit.ts

```diff
@@ -0,0 +1,47 @@
+import { constants } from "@bc/domain/constants";
+import { EntryPoint } from "@bc/domain/entry-point";
+import { Octokit } from "@octokit/rest";
+import Container from "typedi";
+
+/**
+ * Singleton factory for octokit instance
+ */
+export class OctokitFactory {
+    private static octokit?: Octokit;
+
+    /**
+     * Get the initialized octokit instance and initialize it if not done yet.
+     * During initialization, add token to the octokit instance if it is being used 
+     * in a Github runner env
+     * @returns {Octokit} a hydrated octokit instance
+     */
+    public static getOctokitInstance(): Octokit {
+        if (!this.octokit) {
+            const entryPoint: EntryPoint = Container.get(constants.CONTAINER.ENTRY_POINT);
+            switch (entryPoint) {
+                case EntryPoint.CLI:
+                    this.octokit = new Octokit({
+                        userAgent: "kiegroup/github-build-chain-action"
```

Comment:
well token can be specified from CLI entry right? :thinking:

Reply:
that is true